### PR TITLE
Implemented compression flags system table

### PR DESCRIPTION
**Pull request by chands10**
Signed-off-by: Salil Chandra <schandra107@bloomberg.net>

To help us review your pull request, please consider providing an overview of the following:
* What is the type of the change (bug fix, feature, documentation and etc.) ?
* What are the current behavior and expected behavior, if this is a bugfix ?
* What are the steps required to reproduce the bug, if this is a bugfix ?
* What is the current behavior and new behavior, if this is a feature change or enhancement ?
* [Optional] Why is the new behavior better than the current behavior, if this is a feature change ?

### CODE REVIEW

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
```

File: sqlite/ext/comdb2/table_properties.c [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    const char *odh;
+    const char *compress;
+    const char *blob_compress;
+    const char *in_place_updates;
+    const char *instant_schema_change;
+} systable_table_properties_t;
+
+int table_properties_systable_collect(void **data, int *nrecords)
+{
+    rdlock_schema_lk();
```
Comment (riverszhang89) [inline sqlite/ext/comdb2/table_properties.c:41]: You can't simply grab schema_lk here. The thread might have already held berkdb locks and thus may deadlock with a schema change. Consider using "comdb2_systable" table lock instead (which is a berkdb lock, too). See [this example](https://github.com/bloomberg/comdb2/blob/master/sqlite/ext/comdb2/columns.c#L320).

**Review by roborivers — COMMENTED** [review]
Lint check:

```diff


@@ sqlite/ext/comdb2/tables.c @@
+ If a new system table is added, be sure to document it in docs/pages/programming/system_tables.md.
```

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>105/409 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>auth_twofiles_generated<br>auth_queueodh_generated<br>auth<br>userschema_queueodh_generated<br>userschema<br>comdb2sys_pagesize_generated<br>comdb2sys_queueodh_generated<br>comdb2sys<br>cdb2sql_queueodh_generated<br>cdb2sql</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/409 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>comdb2sys_pagesize_generated [setup failure]<br>socksql_master_swings<br>sc_transactional_rowlocks_generated<br>tunables</p>

**Pull request merged**